src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+---
+title: Understanding your UI as a Tree 
+---
+
+<Intro>
+
+Your React app is taking shape with many components being nested within each other. How does React keep track of your app's component structure?
+
+React, and many other UI libraries, model UI as a tree. Thinking of your app as a tree is useful for understanding the relationship between components to debug future concepts like performance and state management.
+
+</Intro>
+
+<YouWillLearn>
+
+* How React "sees" component structures
+* What a render tree is and what it is useful for
+* What a module dependency tree is and what it is useful for
+
+</YouWillLearn>
+
+## Your UI as a tree {/*your-ui-as-a-tree*/}
+
+Trees are a relationship model between items and UI is often represented using tree structures. For example on browsers, the [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model/Introduction) represents HTML elements while the [CSSOM](https://developer.mozilla.org/docs/Web/API/CSS_Object_Model) does the same for CSS. Similar for mobile, platform views relate to one another in a tree format. There’s even an [Accessibility tree](https://developer.mozilla.org/docs/Glossary/Accessibility_tree)!
+
+<Diagram name="preserving_state_dom_tree" height={193} width={864} alt="Diagram with three sections arranged horizontally. In the first section, there are three rectangles stacked vertically, with labels 'Component A', 'Component B', and 'Component C'. Transitioning to the next pane is an arrow with the React logo on top labeled 'React'. The middle section contains a tree of components, with the root labeled 'A' and two children labeled 'B' and 'C'. The next section is again transitioned using an arrow with the React logo on top labeled 'React'. The third and final section is a wireframe of a browser, containing a tree of 8 nodes, which has only a subset highlighted (indicating the subtree from the middle section).">
+
+From components, React creates a UI tree which React DOM uses to render the DOM
+</Diagram>
+
+React also uses tree structures to manage and model the relationship between components in a React app. These trees are useful tools to understand how data flows through a React app and how to optimize rendering and app size.
```

Comment:
Might be nice to have a note that DOM is not the only thing you can render to e.g. mobile, desktop, VR, etc.

Reply:
I mention this in the deep dive. I think it might make more sense there -- at this point, I think I'd like to keep the emphasis that "trees are useful data structures" and "React uses trees just like browsers, mobile platforms, etc". Rendering is not necessarily related to the concept of a React tree.